Assets/General/Scripts/MenuFunctionality/AudioSettings.cs

```diff
@@ -0,0 +1,169 @@
+    {
+        // Set the VCA to the correct volume to change
+        _vca = FMODUnity.RuntimeManager.GetVCA("vca:/MasterVCA");
+        _masterVolume = _masterSlider.value;
+        // Set the volume and save the data
+        _vca.setVolume(_masterVolume);
+        SaveData();
+    }
+
+    /// <summary>
+    /// Change the sfx volume to the corresponding slider
+    /// </summary>
+    public void ChangeSfx()
+    {
+        // Set the VCA to the correct volume to change
+        _vca = FMODUnity.RuntimeManager.GetVCA("vca:/SFXVCA");
+        _sfxVolume = _sfxSlider.value;
+        // Set the volume and save the data
+        _vca.setVolume(_sfxVolume);
+        SaveData();
+    }
+
+    /// <summary>
+    /// Change the ambience volume to the corresponding slider
+    /// </summary>
+    public void ChangeAmbience()
+    {
+        // Set the VCA to the correct volume to change
+        _vca = FMODUnity.RuntimeManager.GetVCA("vca:/AmbianceVCA");
+        _ambienceVolume = _ambienceSlider.value;
+        // Set the volume and save the data
+        _vca.setVolume(_ambienceVolume);
+        SaveData();
+    }
+
+    /// <summary>
+    /// Change the voice volume to the corresponding slider
+    /// </summary>
+    public void ChangeVoice()
+    {
+        // Set the VCA to the correct volume to change
+        _vca = FMODUnity.RuntimeManager.GetVCA("vca:/DialogueVCA");
+        _voiceVolume = _voiceSlider.value;
+        // Set the volume and save the data
+        _vca.setVolume(_voiceVolume);
+        SaveData();
+    }
+
+    /// <summary>
+    /// Change the music volume to the corresponding slider
+    /// </summary>
+    public void ChangeMusic()
+    {
+        // Set the VCA to the correct volume to change
+        _vca = FMODUnity.RuntimeManager.GetVCA("vca:/MusicVCA");
+        _musicVolume = _musicSlider.value;
+        // Set the volume and save the data
+        _vca.setVolume(_musicVolume);
+        SaveData();
+    }
+
+    /// <summary>
+    /// Load the save data from the file, converting it to an array
+    /// </summary>
+    /// <returns>The array of all volumes</returns>
+    private float[] LoadData()
+    {
+        // Create an array and populate it with the volumes as strings
+        float[] volumeArray = new float[5];
+        string[] volumeStrings = File.ReadAllLines(Application.streamingAssetsPath + _audioSettingsFilePath)[0].Split(" ");
+
+        // Iterate through each string and convert it to a float
+        for (int i = 0; i < volumeArray.Length; i++)
+        {
+            // If the string is possible to convert to a float, do so
+            try
+            {
+                volumeArray[i] = float.Parse(volumeStrings[i]);
+            }
+            catch
+            {
+                volumeArray[i] = 1;
+            }
+        }
+
+        return volumeArray;
+    }
+
+    /// <summary>
+    /// Save the volumes to the save file
+    /// </summary>
+    private void SaveData()
```

Comment:
I feel this functionality could be done through our Save Manager, but I'm not going to request changes as it's a bit late to be doing so

Reply:
I wasn't even aware we had save functionality until after this was written, I can fix it next semester